src/Controller/ProductController.php

```diff
@@ -0,0 +1,213 @@
+{
+    /**
+     * @var SerializerInterface
+     */
+    private $serializer;
+
+    public function __construct(SerializerInterface $serializer)
+    {
+        $this->serializer = $serializer;
+    }
+
+    /**
+     * @Get(
+     *     path = "/products/{id}",
+     *     name = "product_show_id",
+     *     requirements = {"id": "\d+"}
+     * )
+     * @Get(
+     *     path = "/products/{model}",
+     *     name = "product_show_model"
+     * )
+     */
+    public function getProductAction(Product $product)
+    {
+        $view = $this->view($product, Response::HTTP_OK);
+
+        return $this->handleView($view);
+    }
+
+    /**
+     * @Get(
+     *     path = "/products",
+     *     name = "product_list"
+     * )
+     * @Rest\QueryParam(
+     *     name = "property",
+     *     requirements = "id|price|quantity|brand|model",
+     *     default = "price",
+     *     description = "Sort order (asc or desc)"
+     * )
+     * @Rest\QueryParam(
+     *     name = "order",
+     *     requirements = "asc|desc",
+     *     default = "asc",
+     *     description = "Sort order (asc or desc)"
+     * )
+     * @Rest\QueryParam(
+     *     name = "search",
+     *     default = null,
+     *     description = "Requested brand or model. The query parameter *property* must be either brand or model to work."
+     * )
+     * @Rest\QueryParam(
+     *     name = "exact",
+     *     default = "true",
+     *     description = "If false, every products containing the search parameter will be returned. The search will be based on the property parameter value. If true, only products having the exact value of the search parameter will be returned. "
+     * )
+     * @Rest\QueryParam(
+     *     name = "page",
+     *     requirements = "\d+",
+     *     default = 1,
+     *     description = "Page number"
+     * )
+     * @Rest\QueryParam(
+     *     name = "itemsPerPage",
+     *     requirements = "\d+",
+     *     default = 5,
+     *     description = "Number of items per page"
+     * )
+     * @param ProductRepository $repository
+     * @param string $property
+     * @param string $order
+     * @param string|null $search
+     * @param string $exact
+     * @param int $page
+     * @param int $itemsPerPage
+     * @return Response
+     */
+    public function getProductsAction(
+        ProductRepository $repository,
+        string $property = "price",
+        string $order = "asc",
+        ?string $search = null,
+        string $exact = "true",
+        int $page = 1,
+        int $itemsPerPage = 5
+    ) {
+        if ($search !== null) {
+            if (in_array($property, ["brand", "model"])) {
+                $criteria = [$property => $search];
+            } else {
+                return new Response("Can not use search parameter, the property is either missing or wrong.", Response::HTTP_NOT_ACCEPTABLE);
+            }
+        }
+
+        $exact === "false" ? $exactValue = false : $exactValue = true;
```

Comment:
`$exactValue = $exact !== "false";`

Reply:
Refactored. But what would be the best way to handle this?

I'm using a string here, but at first, I used a parameter `bool $exact` so the query was written like that: `exact=0` for false and `exact=1` for true.

Which one should I use? Or maybe there is a better way than these two?